optaplanner-test/src/main/java/org/optaplanner/test/impl/score/stream/DefaultConstraintVerifier.java

```diff
@@ -56,6 +56,21 @@ public ConstraintStreamImplType getConstraintStreamImplType() {
         return this;
     }
 
+    public boolean isDroolsAlphaNetworkCompilerEnabled() {
+        return droolsAlphaNetworkCompilerEnabled;
+    }
+
+    @Override
+    public ConstraintVerifier<ConstraintProvider_, Solution_>
+            withDroolsAlphaNetworkCompiler(boolean droolsAlphaNetworkCompilerEnabled) {
+        if (getConstraintStreamImplType() != ConstraintStreamImplType.DROOLS && droolsAlphaNetworkCompilerEnabled) {
```

Comment:
I wouldn't even check the value of droolsAlphaNetworkCompilerEnabled.
If this is called and it's not using drools, let's just fail fast?